packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/ServiceExecutionState.ts

```diff
@@ -661,17 +661,32 @@ export abstract class ServicePureExecutionState extends ServiceExecutionState {
         );
       }
     } catch (error) {
-      assertErrorThrown(error);
-      this.editorStore.applicationStore.logService.error(
-        LogEvent.create(GRAPH_MANAGER_EVENT.EXECUTION_FAILURE),
-        error,
-      );
-      this.editorStore.applicationStore.notificationService.notifyError(error);
+      // Don't notify users about success or failure
+      if (this.queryRunPromise === promise) {
+        assertErrorThrown(error);
+        this.editorStore.applicationStore.logService.error(
+          LogEvent.create(GRAPH_MANAGER_EVENT.EXECUTION_FAILURE),
+          error,
+        );
+        this.editorStore.applicationStore.notificationService.notifyError(
+          error,
```

Comment:
You made a comment here that we are not gonna user about the error then we should probably remove this?

Reply:
Heyy, we don't wanna notify users about any failure or success from the query that is being canceled. 
However, we should notify users about any failure from the current query that is being executed.
Therefore, ` if (this.queryRunPromise === promise) ` is used to identify if this failure is from the canceled query or executed query.